src/open_inwoner/openzaak/tests/test_zgw_imports.py

```diff
@@ -188,6 +190,23 @@ def test_import_catalogs(self, m):
             ),
         )
 
+    def test_import_catalogs_updates_service(self, m):
+        data = {root: CatalogMockData(root).install_mocks(m) for root in self.roots}
+        another_service = ServiceFactory()
+        existing_catalogus_with_same_url_but_different_service = CatalogusConfigFactory(
+            service=another_service, url=data[self.roots[0]].catalogs[0]["url"]
+        )
+
+        import_catalog_configs()
+
+        existing_catalogus_with_same_url_but_different_service.refresh_from_db()
+        self.assertEqual(
+            existing_catalogus_with_same_url_but_different_service.service,
+            self.api_groups[0].ztc_service,
+            msg="Existing catalog with matching URL but differing Service should "
+            "be updated",
+        )
```

Comment:
Strictly speaking, we want to test that `existing_catalogus_with_same_url_but_different_service.service` is the same as the *original* `self.api_groups[0].ztc_service`. Your test is compatible with the scenario where `self.api_groups[0].ztc_service` was updated to equal `another_servicve`. The two would still be equal in this case. You could save the original `self.api_groups[0].ztc_service` in a variable before calling `import_catalog_configs` and use it in the assert to exclude this possibility. It's perhaps far-fetched since your code for updating the service is pretty straightforward...